### PR TITLE
TicTacToe game using ERC7715 permissions

### PR DESCRIPTION
**Pull request by KannuSingh**
Implemented TicTacToe Game which utilize the ERC7715 Permissions (async signing of user's action is performed by Dapp server)

### CODE REVIEW

File: advanced/dapps/tic-tac-toe/src/app/api/signer/route.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { NextResponse } from "next/server";
+import { privateKeyToAccount } from "viem/accounts";
+
+export function GET() {
+  try {
+    const APPLICATION_PRIVATE_KEY = process.env
+      .NEXT_PUBLIC_APPLICATION_PRIVATE_KEY as `0x${string}`;
```
Comment (lukaisailovic) [inline advanced/dapps/tic-tac-toe/src/app/api/signer/route.ts:7]: why is this `NEXT_PUBLIC_APPLICATION_PRIVATE_KEY` if its used from an api route?

File: advanced/dapps/tic-tac-toe/src/utils/TicTacToeUtils.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+import { GrantPermissionsParameters, GrantPermissionsReturnType } from 'viem/experimental'
+import { createPublicClient, createWalletClient, decodeEventLog, http,} from 'viem';
+import { privateKeyToAccount } from 'viem/accounts';
+import { sepolia } from 'viem/chains';
+import { getBlockchainApiRpcUrl } from './ChainsUtil';
+import { executeActionsWithECDSAAndCosignerPermissions } from './ERC7715PermissionsAsyncUtils';
+
+export const ticTacToeAddress = '0xC77189Ad823767366b1c5459029CE638ef5AEc12' as `0x${string}`
+const PROJECT_ID = process.env.NEXT_PUBLIC_PROJECT_ID;
+const APPLICATION_PRIVATE_KEY = process.env.NEXT_PUBLIC_APPLICATION_PRIVATE_KEY ;
```
Comment (lukaisailovic) [inline advanced/dapps/tic-tac-toe/src/utils/TicTacToeUtils.ts:12]: Why do we have private key here?

**Comment by lukaisailovic** [comment]
We use API route but seems like we are leaking private keys

File: advanced/dapps/tic-tac-toe/src/app/api/signer/route.ts (hunk 1 above)
Comment (KannuSingh) [reply advanced/dapps/tic-tac-toe/src/app/api/signer/route.ts:7]: the route is getting the private key from .env.  This route is for getting the dapp signer. 
I didn't get the concern here?

Comment (lukaisailovic) [reply advanced/dapps/tic-tac-toe/src/app/api/signer/route.ts:7]: NEXT_PUBLIC prefixed env is for bowser. You can check the next documentation [here](https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables)

Comment (KannuSingh) [reply advanced/dapps/tic-tac-toe/src/app/api/signer/route.ts:7]: Okay, Understood.

**Pull request merged**